omod/src/main/resources/config.xml

```diff
@@ -22,9 +26,13 @@
 		</init-param>
 	</filter>
 
+    <filter-mapping>
+        <filter-name>oauth2ServiceAccountFilter</filter-name>
+        <url-pattern>/*</url-pattern>
+    </filter-mapping>
```

Comment:
@wluyima why not simply hook yourself into the existing `oauth2-request-filter`?

(And purely on the naming convention side, let's choose here between `camelCase` and `kebab-case` for the filter name, and be consistent across all filters.)

Reply:
I actually had it that way before, the only reason I moved it to a separate filter was purely separation of concerns and ease of testability when I finally add tests so that I test this specific logic